### PR TITLE
Issue 20 - Fix sitemap

### PR DESCRIPTION
**Pull request by rodfersou**

### CODE REVIEW

File: collective/googlenews/browser/sitemap.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    def news(self):
+        """Return last news for Google News sitemap"""
+        portal_types = api.portal.get_registry_record(
+            GoogleNewsSettings.__identifier__ + '.portal_types')
+        catalog = api.portal.get_tool('portal_catalog')
+        brains = catalog(
+            portal_type=portal_types, sort_on='effective',
+            sort_order='reverse', sort_limit=1000,
+            effective={'query': (DateTime() - 2, DateTime()), 'range': 'min:max'})
```
Comment (hvelarde) [inline collective/googlenews/browser/sitemap.py]: effective=dict(query=DateTime() - 2, range='max')

File: collective/googlenews/browser/sitemap.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        self.context = context
+        self.request = request
+
+    def _brain2news(self, brain):
+        """Transform brain into Google News data"""
+        portal = api.portal.get()
+        return {
+            'loc': brain.getURL(),
+            'name': portal.Title(),
+            'language': (
```
Comment (hvelarde) [inline collective/googlenews/browser/sitemap.py]: `language` should be calculated ouside of the loop, probably in the `news` function:

```python
self.language = portal.language if portal.language in ('zh-cn', 'zh-tw') else portal.language[:2]

...

'language': self.language,
```

File: collective/googlenews/browser/sitemap.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+    def _brain2news(self, brain):
+        """Transform brain into Google News data"""
+        portal = api.portal.get()
+        return {
+            'loc': brain.getURL(),
+            'name': portal.Title(),
```
Comment (hvelarde) [inline collective/googlenews/browser/sitemap.py]: `name` should be set outside of the loop, probably in the `news` function, and there's no need to include it in the list.

File: collective/googlenews/tests/test_sitemap.py [hunk 4]
```diff
@@ -1,33 +1,23 @@
...
-        self.view = sitemap.GoogleNewsSiteMap(self.context, self.request)
-        self.view.portal_state = utils.FakePlonePortalState()
-        self.view.settings = utils.FakeSettings()
-        self.view.update()
+        self.portal = self.layer['portal']
+        self.request = self.layer['request']
+        self.view = getMultiAdapter(
+            (self.portal, self.request), name='googlenews-sitemap.xml')
 
     def test_brain2news(self):
```
Comment (hvelarde) [inline collective/googlenews/tests/test_sitemap.py]: just remove this as it's not used

File: collective/googlenews/browser/sitemap.py (hunk 1 above)
Comment (rodfersou) [inline collective/googlenews/browser/sitemap.py]: effective=dict(query=DateTime() - 2, range='min') ;)

**Pull request merged**